File: application/spring/src/main/java/tn/esprit/stage_ey/Entities/CartItem.java
package tn.esprit.stage_ey.Entities;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class CartItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private int quantity;


    @ManyToOne
    private Product productt;

    @ManyToOne
    private Cart cartt;



    // Getters and setters
}

File: application/angular/src/app/Entity/CartItem.ts
import {Cart} from "./Cart";
import {Product} from "./Product";

export class CartItem {
  id: number;
  product: Product;
  cart: Cart;
  quantity: number;

  constructor(id: number, product: Product, cart: Cart, quantity: number) {
    this.id = id;
    this.product = product;
    this.cart = cart;
    this.quantity = quantity;
  }
}
